### DISCUSSION TITLE
How can i mark a field as selected on my edit page in filament ?

### DISCUSSION THREAD

**Discussion by chahal26** [filamentphp/filament#2909] [Help] [answered]
I need to mark an option as selected on my edit page in the filament. 

Also, I am using  `Category::get()->pluck('name', 'id')`  in options. How can I remove a particular field using where condition based on another field on the edit page.

For example, can we do something like  `Category::where('parent_id','!=',$record->id)->get()->pluck('name', 'id')` ? 

Hope this makes sense.

**Comment by studiowizjo** [comment, marked as answer]
> I need to mark an option as selected on my edit page in the filament.

Field name should match your attribute name and then it should work.

> Also, I am using Category::get()->pluck('name', 'id') in options. How can I remove a particular field using where condition based on another field on the edit page.

Use callback and pass to it $record or $get:

```php
->options(fn(?Category $record) => Category::where('parent_id', '!=', $record->id)->pluck('name', 'id'))

//or:

->options(fn($get) => Category::where('parent_id', '!=', $get('another_field_from_form'))->pluck('name', 'id'))
```

**Reply by chahal26** [reply]
Hi @studiowizjo, thanks for your reply. It worked fine on the edit page but it is throwing an error on add page as we don't want any condition there. On add page, it should be 

`Category:pluck('name', 'id'))`

**Reply by danharrin** [reply]
Check if `$record` is `null` first then?

**Reply by chahal26** [reply]
Thanks @danharrin, that worked.
